src/query.js

```diff
@@ -1,335 +1,708 @@
 
-/** See the [docs](https://faunadb.com/documentation/queries#auth_functions). */
-export function identify(ref, password) {
-  return {identify: ref, password}
+/**
+ * See the [docs](https://faunadb.com/documentation/queries#auth_functions).
+ *
+ * @param {module:query~ExprArg} ref
+ * @param {module:query~ExprArg} password
+ * @return {Expr}
+ */
+function Identify(ref, password) {
+  return new Expr({ identify: Expr.wrap(ref), password: Expr.wrap(password) });
 }
 
 // String functions
 
-/** See the [docs](https://faunadb.com/documentation/queries#string_functions). */
-export function concat(strings, separator = null) {
-  return params({concat: strings}, {separator})
+/**
+ * See the [docs](https://faunadb.com/documentation/queries#string_functions).
+ *
+ * @param {module:query~ExprArg} strings
+ * @param {?module:query~ExprArg} separator
+ * @return {Expr}
+ */
+function Concat(strings, separator) {
+  separator = defaults(separator, null);
+  return new Expr(params({ concat: Expr.wrap(strings) }, { separator: Expr.wrap(separator) }));
 }
 
-/** See the [docs](https://faunadb.com/documentation/queries#string_functions). */
-export function casefold(string) {
-  return {casefold: string}
+/**
+ * See the [docs](https://faunadb.com/documentation/queries#string_functions).
+ *
+ * @param {module:query~ExprArg} string
+ * @return {Expr}
+ */
+function Casefold(string) {
+  return new Expr({ casefold: Expr.wrap(string) });
 }
 
 // Time and date functions
-
-/** See the [docs](https://faunadb.com/documentation/queries#time_functions). */
-export function time(string) {
-  return {time: string}
+/**
+ * See the [docs](https://faunadb.com/documentation/queries#time_functions).
+ *
+ * @param {module:query~ExprArg} string
+ * @return {Expr}
+ */
+function Time(string) {
+  return new Expr({ time: Expr.wrap(string) });
 }
 
-/** See the [docs](https://faunadb.com/documentation/queries#time_functions). */
-export function epoch(number, unit) {
-  return {epoch: number, unit}
+/**
+ * See the [docs](https://faunadb.com/documentation/queries#time_functions).
+ *
+ * @param {module:query~ExprArg} number
+ * @param {module:query~ExprArg} unit
+ * @return {Expr}
+ */
+function Epoch(number, unit) {
+  return new Expr({ epoch: Expr.wrap(number), unit: Expr.wrap(unit) });
 }
 
-/** See the [docs](https://faunadb.com/documentation/queries#time_functions). */
-export function date(string) {
-  return {date: string}
+/**
+ * See the [docs](https://faunadb.com/documentation/queries#time_functions).
+ *
+ * @param {module:query~ExprArg} string
+ * @return {Expr}
+ */
+function Date(string) {
+  return new Expr({ date: Expr.wrap(string) });
 }
 
 // Miscellaneous functions
 
-/** See the [docs](https://faunadb.com/documentation/queries#misc_functions). */
-export function equals(...values) {
-  return {equals: varargs(values)}
+/**
+ * See the [docs](https://faunadb.com/documentation/queries#misc_functions).
+ *
+ * @return {Expr}
+ */
+function NextId() {
```

Comment:
`NextID`? My intuition is to prefer consistency across clients if applicable, but it's not a big deal.